### DISCUSSION TITLE
Experimental netcdf4 interface package

### DISCUSSION THREAD

**Discussion by LKedward** [fortran-lang/fpm#458] [Show and tell]
I've mentioned in previous threads that lightweight packages containing module interfaces are a better way of interfacing with system-installed packages instead of relying on `.mod` files. I've put together such a package for netcdf-fortran here: <https://github.com/LKedward/netcdf-interfaces>.

Reading through the structure of [netcdf-fortran](https://github.com/Unidata/netcdf-fortran) I came to the tentative conclusion that the existing modern fortran interface modules therein are already well-defined and self-contained, and therefore sufficient for use as an fpm package that interfaces with an existing netcdf installation.

Moreover since the modern interface module only calls an `f77` style API (I think), then this *fpm* package should be useable across different compilers, even if `netcdf-fortran` itself has only been compiled with a single compiler. The package currently contains some test programs copied from the netcdf-fortran repo, which run successfully in the CI with both `gfortran` and `ifort`  (where netcdf-fortran is always compiler with `gfortran`).

The only caveat so far is that you need to use `--flag -fallow-argument-mismatch` with `gfortran-10`.
If anyone has success (or not) using it with their projects, do let me know.

**Comment by milancurcic** [comment]
Fantastic, I will try it in the coming week.

How do you think can this be best kept up to date with future NetCDF versions? If I understand it right, the interface should keep working as long as the API doesn't change. How do we update the interface when it does change? Is it possible to keep it in sync with the upstream repo?

**Reply by LKedward** [reply]
Yes, that it my understanding also. Versioning and keeping it up-to-date is another caveat; it should be possible to use a github actions cron job to regenerate the package if the upstream copy changes. My *hope* is that the API is fairly stable at this point and so synchronised versioning can be done manually, perhaps by generating the package and tagging the releases to coincide with with major releases of netcdf-fortran.
